File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        len1=len(haystack)
        len2=len(needle)
        i=0

        while i <= len1-len2:
            j=0
            while(j<len2 and haystack[i+j]==needle[j]):
                j+=1
            if(j== len2):
                return i
            i+=1
        
        return -1

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.c
int strStr(char* haystack, char* needle) {
    int len1=strlen(haystack);
    int len2 = strlen(needle);
    int i=0;int j=0;
    if(len1<len2){
        return -1;
    }
    while(i<=len1-len2){
        j=0;
        while(j<len2&&haystack[i+j]==needle[j]){
            j++;
        }
        if(j==len2){
            return i;
        }
        i++;
    }

    return -1;


}